### PR TITLE
[BEAM-8019] Several changes to support cross-language transforms on DataflowRunner

### PR DESCRIPTION
**Pull request by chamikaramj**
Contains following changes. Didn't break these up into separate PRs since these are small changes but lemme know if I should.

* Python SDK's current implementation requires one output to be identified as the main output. But external transforms do not identify a main output and I don't think this is required for execution of the transforms. So updated the code to pick one output tag of external transforms to be the main output tag.
* Updates Python SDK to not to override the main output tag for external ParDo transforms. Java SDK in some cases requires the main output tag of ParDo transforms to not to change to operate correctly.
* Updated TestPipeline to be able to provide additional pipeline options. This is is helpful when integration testing cross-language pipelines.
* Fixed a bug when comparing environments in RunnerAPIPTransformHolder.to_runner_api() method.
 
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @robertwb PTAL

File: sdks/python/apache_beam/pipeline.py [hunk 1]
```diff
@@ -917,8 +917,20 @@ def is_side_input(tag):
         for tag, id in proto.outputs.items()}
     # This annotation is expected by some runners.
     if proto.spec.urn == common_urns.primitives.PAR_DO.urn:
-      result.transform.output_tags = set(proto.outputs.keys()).difference(
-          {'None'})
+      # Some external transforms require output tags to be not to be modified.
+      # So we randomly select one of the output tags as the main output and
+      # leave others as side outputs. Transform execution should not change
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/pipeline.py]: Looking at the code some more, output_tags is only ever used by the dataflow runner. Could we please isolate these workarounds there instead and simply delete the output_tags attribute here (and also not introduce main_output_tag)?

**Comment by chamikaramj** [comment]
Run Python 2 PostCommit

File: sdks/python/apache_beam/testing/test_pipeline.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def __init__(self,
                options=None,
                argv=None,
                is_integration_test=False,
-               blocking=True):
+               blocking=True, additional_pipeline_args=None):
```
[the remark this answers is not in the document]
Comment (chamikaramj) [reply sdks/python/apache_beam/testing/test_pipeline.py]: Done.

**Comment by chamikaramj** [comment]
Thanks. PTAL.

**Comment by chamikaramj** [comment]
Thanks. Will merge after tests pass.

**Comment by chamikaramj** [comment]
Run Python PreCommit

**Comment by chamikaramj** [comment]
Run Python PreCommit

**Pull request merged**